virttest/utils_misc.py

```diff
@@ -2933,71 +2969,117 @@ def format_linux_disk(session, did, all_disks_did, partition=False,
                           disk kname, serial and wwn.
     :param partition: if true, can format all disks; otherwise,
                       only format the ones with no partition originally.
+    :param n_partitions: the number of partitions on disk
     :param mountpoint: mount point for the disk.
-    :param size: partition size.
+    :param size: partition size. e.g. 2G
     :param fstype: filesystem type for the disk.
-    :return Boolean: disk usable or not.
-    """
-    disks = list_linux_guest_disks(session, partition)
-    logging.debug("Disks detail: %s" % disks)
-    for line in disks:
-        kname = line.split('/')[2]
-        did_list = all_disks_did[kname]
-        if did not in did_list:
-            # Continue to search target disk
+    :param labeltype: label type for the disk.
+    :return a list: mount point list for all partitions.
+    """
+    if not all_disks_did:
+        all_disks_did = get_linux_disks_did(session, partition).values()
+    for disk_did in all_disks_did:
+        if did not in disk_did:
             continue
+        kname = disk_did[0]
         if not size:
-            size_output = session.cmd_output_safe("lsblk -oKNAME,SIZE|grep %s"
-                                                  % kname)
-            size = size_output.splitlines()[0].split()[1]
-        all_disks_before = list_linux_guest_disks(session, True)
-        devname = line
-        logging.info("Create partition on disk '%s'" % devname)
-        mkpart_cmd = "parted -s %s mklabel gpt mkpart "
-        mkpart_cmd += "primary 0 %s"
-        mkpart_cmd = mkpart_cmd % (devname, size)
-        session.cmd_output_safe(mkpart_cmd)
-        session.cmd_output_safe("partprobe %s" % devname)
-        all_disks_after = list_linux_guest_disks(session, True)
-        partname = (all_disks_after - all_disks_before).pop()
-        logging.info("Format partition to '%s'" % fstype)
-        format_cmd = "yes|mkfs -t %s %s" % (fstype, partname)
-        session.cmd_output_safe(format_cmd)
-        if not mountpoint:
-            session.cmd_output_safe("mkdir /mnt/%s" % kname)
-            mountpoint = os.path.join("/mnt", kname)
-        logging.info("Mount the disk to '%s'" % mountpoint)
-        mount_cmd = "mount -t %s %s %s" % (fstype, partname, mountpoint)
-        session.cmd_output_safe(mount_cmd)
-        return True
+            size = disk_did[1]
+        if size[-1].upper() not in ['B', 'K', 'M', 'G', 'T']:
+            logging.error("Invalid size value: '%s'" % size)
+            return []
+        devname = '/dev/%s' % kname
+        partition_list = []
+        mountpoint_list = []
+        if partition:
+            partition_list.append(kname)
+        else:
+            logging.info("Create partition on disk %s" % devname)
+            partition_size = int(size[0:-1])/n_partitions
```

Comment:
partition_size should be a float, if size is 1G and n_partitions is 2, partition_size will get 0.

Reply:
updated, thanks